### PR TITLE
[18.0][MIG] partner_identification: Migration to 18.0

### PR DESCRIPTION
**Pull request by chaule97**
**OCA - port**

- https://github.com/OCA/partner-contact/pull/1853

### CODE REVIEW

File: test-requirements.txt
Comment (TDu) [inline test-requirements.txt:1]: Why the `DON'T MERGE` in the commit message ?

File: partner_identification/models/res_partner.py
Comment (TDu) [inline partner_identification/models/res_partner.py]: Any good reason to replace format, OCA recommendation maybe ? If not I would not include this change in the migration.
But I may be missing something

File: test-requirements.txt
Comment (chaule97) [reply test-requirements.txt:1]: this module requires `odoo_test_helper` lib to run [testcase](https://github.com/OCA/partner-contact/blob/e1cff143a9e7fd7927e8a1394731f9a21aa3051e/partner_identification/tests/test_res_partner.py#L4) but odoo requirements.txt does not have `odoo_test_helper` lib

File: partner_identification/models/res_partner.py
Comment (chaule97) [reply partner_identification/models/res_partner.py]: yes, OCA recommend it. we need to change to pass pre-commit

Comment (pedrobaeza) [reply partner_identification/models/res_partner.py]: Where it's recommended?

Comment (nilshamerlinck) [reply partner_identification/models/res_partner.py]: since [odoo 14.0](https://github.com/odoo/odoo/pull/52155), `_()` has support for placeholders, so here the good move would be to change
```
                    % {
                        "record_name": record._name,
                        "category_code": category_code,
                        "field_name": field_name,
                    }
 ```

to

```            
record_name=record._name,
category_code=category_code,
field_name=field_name
```

Comment (chaule97) [reply partner_identification/models/res_partner.py]: I'm sorry everyone because I don't read carefully. I have updated it

File: test-requirements.txt
Comment (TDu) [reply test-requirements.txt:1]: Yes the change is correct, and we want it to be merged.
But the title of the commit needs to be changed by removing `[DON'T MERGE]`
This convention is used when we need to add a temporary reference to an existing PR more information in https://github.com/OCA/maintainer-tools/wiki/Use-temporary-reference(s)-to-another-pull-request(s)

**Review by TDu — COMMENTED** [review]
Functional test on runbot LG.
Still needs to change that commit message

File: partner_identification/models/res_partner.py
Comment (TDu) [reply partner_identification/models/res_partner.py]: > since [odoo 14.0](https://github.com/odoo/odoo/pull/52155), `_()` has support for placeholders, so here the good move would be to change
> 
> ```
>                     % {
>                         "record_name": record._name,
>                         "category_code": category_code,
>                         "field_name": field_name,
>                     }
> ```
> 
> to
> 
> ```
> record_name=record._name,
> category_code=category_code,
> field_name=field_name
> ```

Thanks for the link, I agree

**Comment by chaule97** [comment]
> Still needs to change that commit message

Excuse me, I don't understand

**Comment by chaule97** [comment]
> Functional test on runbot LG. Still needs to change that commit message

Hi, I have changed commit message

**Review by TDu — APPROVED** [review]

**Review by simahawk — APPROVED** [review]
LG

**Comment by chaule97** [comment]
ciao @simahawk , I have rebased branch 18.0 and pre-commit again. Can you merge again

**Pull request merged**
